### PR TITLE
Add parameters validation on token upload endpoint

### PR DESCRIPTION
**Pull request by mshaposhnik**
### What does this PR do?
Add namespace and token name parameters validation on token upload endpoint.
- Namespaces must comply with DNS1123 labels requirement. (see https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#namespaces-and-dns)
- Tokens must comply general label values requirement, since the SPI access token names are used as a
label values in the linked objects.

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-364

### How to test this PR?

```
quay.io/redhat-appstudio/pull-request-builds:spi-operator-#
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-#
```
Perform a token upload POST with invalid parameters.  Expect `400` response with the explanation.

### CODE REVIEW

File: oauth/handlers.go [hunk 1]
```diff
@@ -98,6 +100,21 @@ func HandleUpload(uploader TokenUploader) func(http.ResponseWriter, *http.Reques
 			return
 		}
 
+		// We want to check token names are satisfying the general label value format,
+		// since the SPI access token names are used as a label values in the linked objects.
+		errs := validation.IsValidLabelValue(tokenObjectName)
+		if len(errs) > 0 {
+			LogDebugAndWriteResponse(r.Context(), w, http.StatusBadRequest, "Incorrect token name parameter. Must comply common label value format. Details: "+strings.Join(errs, ";"))
```
Comment (metlos) [inline oauth/handlers.go:107]: ... comply with ...

File: oauth/handlers.go [hunk 2]
```diff
@@ -98,6 +100,21 @@ func HandleUpload(uploader TokenUploader) func(http.ResponseWriter, *http.Reques
...
+			return
+		}
+
+		// Namespaces are required to be a valid RFC 1123 DNS labels
+		errs = validation.IsDNS1123Label(tokenObjectNamespace)
+		if len(errs) > 0 {
+			LogDebugAndWriteResponse(r.Context(), w, http.StatusBadRequest, "Incorrect token namespace parameter. Must comply RFC 1123 label format. Details: "+strings.Join(errs, ";"))
```
Comment (metlos) [inline oauth/handlers.go:114]: ... comply with ...

**Review by metlos — COMMENTED** [review]
LGTM apart from the grammar nitpicks.

Also, not sure if the caller of the OAuth endpoint really needs to know why the name is incorrect. This is basically a check that the names can
be valid at all to save us a call to K8s API that would tell us the same thing, too.

**Review by sparkoo — APPROVED** [review]

**Review by xbaran4 — APPROVED** [review]
Works for me: 
image

**Review by metlos — APPROVED** [review]

**Pull request merged**